lib/ui/common/ActionSheet.dart

```diff
@@ -0,0 +1,203 @@
+  final Radius dropLineRadius = const Radius.circular(2.5);
+
+  final double dropLineHeight = 5;
+  final double cardHeight = 70;
+  final double iconLineSpace = 21.5;
+  final double cardElevation = 0;
+  final double bigIconHeight = 24;
+  final double smallIconHeight = 16.5;
+
+  final int maxLines = 1;
+
+  const DefaultStyle();
+}
+
+class ActionSheet {
+  static Widget build(BuildContext context, {ActionSheetStyle style = const DefaultStyle()}) {
+
+    // FIXME: This would be injected in the call function. Every screen which need an ActionSheet has to have this CustomActions defined in its view model
+    ActionListItemViewModel recordSale = ActionListItemViewModel("Record a new Sale", null, false, icon: "assets/icons/detail_icon_cost.png");
+    ActionListItemViewModel recordCost = ActionListItemViewModel("Record a new Cost", null, false, icon: "assets/icons/flag_kenya.png", selectionIcon: "assets/icons/radio_button_active.png");
+    ActionListItemViewModel testing = ActionListItemViewModel("Record a test", null, true);
+    ActionSheetViewModel viewModel = ActionSheetViewModel([recordSale, recordCost, testing], "Cancel");
+
+    /* FIXME: To make the modal bottom sheet autoresizable (without limit) needs to modify native file bottom_sheet
+        @override
+        BoxConstraints getConstraintsForChild(BoxConstraints constraints) {
+          return BoxConstraints(
+            minWidth: constraints.maxWidth,
+            maxWidth: constraints.maxWidth,
+            minHeight: 0.0,
+            maxHeight: constraints.maxHeight,
+          );
+  }*/
+
+    return Container(
+      color: style.cornersColor, // This line set the transparent background
+        child: Container(
+          decoration: BoxDecoration(
+              color: style.backgroundColor,
+              borderRadius: BorderRadius.only(
+                  topLeft: style.borderRadius,
+                  topRight: style.borderRadius
+              )
+          ),
+          child: HeaderAndFooterListView.builder(
+              physics: NeverScrollableScrollPhysics(),
+              shrinkWrap: true,
+              itemCount: viewModel.actions.length,
+              itemBuilder: (BuildContext context, int index) =>
+            buildActionCell(style, viewModel.actions[index], viewModel.actions.length, index),
+            header: buildDropLine(style),
+            footer: RoundedButton.build(style: ActionSheetLargeRoundedButtonStyle(), context: context, title: viewModel.buttonTitle)
+          )
+    ));
+  }
+
+  static Widget buildDropLine(ActionSheetStyle style) {
+    return Container(
+      height: style.dropLineHeight,
+      margin: style.dropLineEdge,
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.all(style.dropLineRadius),
+        color: style.dropLineColor,
+      ),
+    );
+  }
+
+  static Widget buildActionCell(ActionSheetStyle style, ActionListItemViewModel viewModel, int numberOfActions, int currentAction) {
```

Comment:
Are you using it outside this file? Should be private?
```suggestion
  static Widget _buildActionCell(ActionSheetStyle style, ActionListItemViewModel viewModel, int numberOfActions, int currentAction) {
```

Reply:
Done